res/games/gameitem.css

```diff
@@ -0,0 +1,138 @@
+        {% endif %}
+
+        {% if teams[1] is not defined %}
+            {% set alignment = 'left' %}
+            {% if "-" not in player.login %}
+	            {% set width = 0 %}
+	        {% else %}
+	            {% set width = 130 %}
+	        {% endif %}
+        {% else %}
+            {% set alignment = 'right' %}
+	        {% set width = 130 %}
+        {% endif %}
+        <tr>
+        {% if align == "left" -%}
+            <td align={{ alignment }} width='{{ width }}'>
+            {{ player_login }}
+            </td>
+            <td>
+            <img src='{{ player_country }}'>
+            </td>
+            <td align={{ alignment }} valign='middle' width='10'>
+            {% if player_global_rating is defined %}
+                ({{ player_global_rating }})
+            {% else %}
+                (???)
+            {% endif %}
+            </td>
+        {% else %}
+            <td align='left' valign='middle' width='10'>
+            {% if player_global_rating is defined %}
+                ({{ player_global_rating }})
+            {% else %}
+                (???)
+            {% endif %}
+            </td>
+            <td>
+            <img src='{{ player_country }}'>
+            </td>
+            <td align='left' width={{ width }}>
+            {{ player_login }}
+            </td>
+        {% endif %}
+         </tr>
+{% endmacro -%}
+
+
+{%- macro versus_string(fontsize) %}
+    <td align='middle' valign='middle' height='100%'>
+    <font color ='black' size={{ fontsize }}>VS</font>
+    </td>
+{% endmacro -%}
+
+
+{%- macro modtip(mods) %}
+    <b><br/>With:</b>
+    {% for mod in mods %}
+        <br/> {{ mod }}
+    {% endfor %}
+{% endmacro -%}
+
+
+{%- macro obstip(observers) %}
+    <b>Observers:</b>
+    {% for observer in observers %}
+        <br/>{{ observer.login }} <img src='{{ observer.countrypath }}'> ({{ (observer.global_rating[0]-(observer.global_rating[1]*3))|int }})
+    {% endfor %}
+{% endmacro -%}
+
+
+{%- macro formatting(teams, mods, observers) %}
+    <table border="0" cellpadding="0" cellspacing="5">
+    <tbody>
+    <tr>
+    {% if teams[1] is not defined and teams[0][2] is defined %}
+        <td valign='middle' height='100%'>
+        <font color ='black' size='+5'>FFA</font>
+        </td>
+    {% endif %}
+    <td><table>
+    {% for player in teams[0] %}
+        {{ playermacro(player,align="left") }}
+    {% endfor %}
+    </table></td>
+    {% if teams[1] %}
+        {% if teams[0][4] or teams[1][4] %}
+            {{ versus_string(fontsize='+5') }}
+	    {% elif teams[0][3] or teams[1][3] %}
+	        {{ versus_string(fontsize='+4') }}
+        {% elif teams[0][2] or teams[1][2] %}
+            {{ versus_string(fontsize='+3') }}
+	    {% elif teams[0][1] or teams[1][1] %}
+	        {{ versus_string(fontsize='+2') }}
+        {% else %}
+            {{ versus_string(fontsize='+1') }}
+        {% endif %}
+    {% endif %}
```

Comment:
This code is unreadable.
1. you check `teams[1]` but not `teams[0]`.
2. if/elif/elif/elif/else are written very poor, and are unreadable - bad indentation
3. i see there algoritm, that can be written in one line.
```
{{ versus_string(fontsize='+{}'.format(1 + max([len(teams[0]), len(teams[1])]) ) }}
```
Maybe you'll need to set max value of teams into variable forward.
4. I don't understand, why `fontsize` has value `+number`. How does it work?
5. all time different checks, once you check if "value" is set (`is (not) defined`), another time you check, if value conversion to boolean is positive `{% if teams[1] %}`.

Reply:
1) because teams[0] always exists and is kind of obsolete to check, but i need to check if teams[1] exists because i only need the versus string if there are 2 teams
2)/3) yes, packing up messy code was a thing for later, but thanks a lot for already giving me a good replacement :)
4) dont ask me, ask HTML - it was like that before, and it is working with the plus. i dont know if it  also works without it, i didnt question it as it was that way in the old code too
5) yes, that is kind of messy too, but doesnt really change the outcome, as the teams list is never "false" - its either true or undefined (but i agree, it should be uniformed)

Reply:
2. try to get knowledge, how it works, i don't think, that html needs +1 for font size.

Reply:
http://www.ftls.org/en/tutorial/HTML/index4.shtml says it with "+",
https://www.w3schools.com/tags/att_font_size.asp says it without. i guess both work the same way, good point